apps/meteor/deploy/deploy-docker.sh

```diff
@@ -0,0 +1,267 @@
+    'centos') [[ -n "$(yum -C -e0 -q search docker-compose-plugin)" ]] && run yum install -y docker-compose-plugin || error "failed to install compose v2" ;;
+  esac
+  &>/dev/null docker compose version && { success_msg "compose v2 sucessfully installed"; return; }
+  # well if we're here compose isn't in the repo or something else went wrong; manual install may still work
+  warn "compose v2 not found on the repos...attempting manual install"
+  do_install_compose
+}
+
+post_install_docker() {
+  if am_i_root; then
+    warn "root user detected"
+    info "ignoring docker group"
+  else
+    if groups | grep -qw 'docker'; then
+      success_msg "\"$USER\" is already in the docker group!"
+    else
+      info "adding \"$USER\" to the docker group"
+      run usermod -aG docker $USER
+      run_sudoless newgrp
+    fi
+  fi
+  if [[ $(systemctl is-active docker) == "active" ]]; then
+    success_msg "docker daemon already running!"
+  else
+    info "enabling docker daemon"
+    run systemctl enable --now docker
+  fi
+}
+
+generate_passwords() {
+  if command_exists "openssl"; then
+    random_password() {
+      openssl rand -hex ${1:-26}
+    }
+  else
+    random_password() {
+      head -c${1:-26} /dev/urandom | hexdump -e '/1 "%02x"'
+    }
+  fi
+
+  declare -A secrets=([MONGODB_PASSWORD]="$(random_password)" [MONGODB_ROOT_PASSWORD]="$(random_password)" [MONGODB_REPLICA_SET_KEY]="$(random_password 8)")
+
+  [[ -f .env ]] || touch .env
+
+  info "generating passwords and secret keys for deployment"
+
+  for variable in "MONGODB_ROOT_PASSWORD" "MONGODB_PASSWORD" "MONGODB_REPLICA_SET_KEY"; do
+    _substitute_or_inject_variable "$variable" "${secrets[$variable]}" .env
+  done
+}
+
+deploy() {
+  if ((ENABLE_HTTPS)); then
+    _substitute_or_inject_variable BIND_IP 127.0.0.1 .env
+    read -p "please enter your domain name (without the 'https://'): " domain
+    _substitute_or_inject_variable DOMAIN $domain .env
+    _substitute_or_inject_variable ROOT_URL "https://$domain" .env
+    read -p "please enter your email address for TLS certificate renewal: " email
+    _substitute_or_inject_variable LETSENCRYPT_EMAIL $email .env
+  fi
+
+  info "pulling docker images"
+  run_sudoless docker compose -f ${COMPOSE_TEMPLATE[name]} pull || error "failed to pull some images; see above for more information"
+  ((ENABLE_HTTPS)) && { run_sudoless docker compose -f ${COMPOSE_TRAEFIK_TEMPLATE[name]} pull || error "failed to pull traefik image; see above for more information"; }
+
+  info "deploying Rocket.Chat"
+  run_sudoless docker compose -f ${COMPOSE_TEMPLATE[name]} up -d || error "Rocket.Chat deployment failed; see above for more information"
+  ((ENABLE_HTTPS)) && { run_sudoless docker compose -f ${COMPOSE_TRAEFIK_TEMPLATE[name]} up -d || error "traefik deployment failed; see above for more information"; }
+}
+
+
+main() {
```

Comment:
This is something I talked to @geekgonecrazy about, essentially the purpose is to make it *easier* to use docker. Bulk of everything is being done by docker's official script. Makes it easier to handle the mongo authentication part (is required by sec team, or will be), cloud will be able to use this as its docker deployment snippet, compose dependencies can be handled without making the deployment process longer. 

That said, I have refactoring rocketchatctl in my backlog and have been thinking about adding a `--docker` flag to it. Not really sure if it fits there, since the purpose for that is native install. We could add the --docker there and we won't have to do this here. 

Thoughts?

Reply:
Resolved internally.